x-pack/qa/full-cluster-restart/src/test/java/org/elasticsearch/xpack/restart/FullClusterRestartIT.java

```diff
@@ -59,6 +62,22 @@
     @Before
     public void waitForMlTemplates() throws Exception {
         XPackRestTestHelper.waitForMlTemplates(client());
+        // TODO prevents 6.x warnings; remove in 8.0
+        if (isRunningAgainstOldCluster()) {
```

Comment:
Same note as above about the checking if the version is in range and asserting the index compatible version.